lib/api/controllers/remoteActions/swagger.js

```diff
@@ -0,0 +1,145 @@
+var
+  fs = require('fs'),
+  _ = require('lodash'),
+  path = require('path'),
+  InternalError = require('kuzzle-common-objects').Errors.internalError,
+  q = require('q'),
+  stringify = require('json-stable-stringify'),
+  jsonToYaml = require('json2yaml'),  
+  generateSwagger,
+  writeFiles;
+
+generateSwagger = (routes, apiVersion) => {
+  var swagger;
+
+  swagger = {
+    basePath: '/api',
+    consumes: [
+      'application/json'
+    ],
+    host: 'sandbox.kuzzle.io:7511',
+    info: {
+      contact: {
+        email: 'hello@kuzzle.io',
+        name: 'Kuzzle team',
+        url: 'http://kuzzle.io'
+      },
+      description: 'The Kuzzle REST API',
+      license: {
+        name: 'Apache 2',
+        url: 'http://opensource.org/licenses/apache2.0'
+      },
+      title: 'Kuzzle API',
+      version: apiVersion
+    },
+    produces: [
+      'application/json'
+    ],
+    schemes: [
+      'http'
+    ],
+    swagger: '2.0',    
+    paths: {}
+  };
+
+  routes.forEach(route => {
+    var 
+      reg = /:([^\/]*)/g, 
+      m;
+
+    if (swagger.paths[route.url] === undefined) {
+      swagger.paths[route.url] = {};
+    }
+
+    if (swagger.paths[route.url][route.verb] === undefined) {
+      if (route.infos !== undefined) {
+        route.infos.description += '\nController: ' + route.controller + '. Action: ' + route.action + '.';
```

Comment:
mhh, I could agree with your approach if we piped directly the route.infos to swagger without altering it (in that case, swagger would claim the missing attributes).

But here, we alter the routes.infos.description attribute, adding controller and action.
Then, if the user do not give this attribute, he won't have an explicit error, but just generate a bad description content, and may not be notified about that.